### DISCUSSION TITLE
esp8266 3.0.0 compatibility

### DISCUSSION THREAD

**Discussion by freiscardillo** [Jeffin28/AWS-IOT-CORE-ESP8266#5] [General]
Hi,

I´m trying to compile your code on WEMOS D1 Mini Core 3.0.0 , is there any newer version since on this core I got some Error´s, on 2.4.1 CORE is working fine !

C:\Users\Fabio\Documents\Arduino\libraries\WebSockets\src\WebSocketsClient.cpp: In member function 'void WebSocketsClient::loop()':
C:\Users\Fabio\Documents\Arduino\libraries\WebSockets\src\WebSocketsClient.cpp:173:30: error: 'class BearSSL::WiFiClientSecure' has no member named 'setCACert'
  173 |                 _client.ssl->setCACert((const uint8_t *)_CA_cert, strlen(_CA_cert) + 1);
      |                              ^~~~~~~~~
C:\Users\Fabio\Documents\Arduino\libraries\WebSockets\src\WebSocketsClient.cpp: In member function 'void WebSocketsClient::connectedCb()':
C:\Users\Fabio\Documents\Arduino\libraries\WebSockets\src\WebSocketsClient.cpp:759:26: error: 'class BearSSL::WiFiClientSecure' has no member named 'verify'
  759 |         if(!_client.ssl->verify(_fingerprint.c_str(), _host.c_str())) {
      |                          ^~~~~~
exit status 1
Erro compilando para a placa LOLIN(WEMOS) D1 R2 & mini

**Comment by Jeffin28** [comment]
2.4.2 is ok,try it

**Comment by Jeffin28** [comment]
I will try in WEMOS D1 Mini core, but the code made for the nodemcu 1.0 board manager may be a chance for library issues. You could try it on changing the board manager to NODEMCU 1.0

**Comment by Jeffin28** [comment]
I think your version of WebSockets library have some issues with with the board manager version.

**Comment by freiscardillo** [comment]
I was tritng to implement wifimanager, but had some issues on version 2.4.1, yes you are right your code works fine on 2.4.1, show be nice to have an update to latest core version.

**Reply by Jeffin28** [reply]
I think the wifimanager has a better response on 2.4.2
